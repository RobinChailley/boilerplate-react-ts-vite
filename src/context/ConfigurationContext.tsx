import { ConfigContextValueType } from '@context/configurationContextTypes';
import getConfigContextValue from '@context/getConfigurationContextValue';
import { createContext, useContext } from 'react';

export const ConfigContext = createContext<ConfigContextValueType>(getConfigContextValue());

export const useConfig = () => useContext(ConfigContext);
