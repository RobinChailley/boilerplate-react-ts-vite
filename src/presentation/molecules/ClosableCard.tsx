import { CloseIcon } from '@assets/html-svg';
import { useConfig } from '@context/ConfigurationContext';
import Badge from '@presentation/atoms/Badge';
import React from 'react';

interface ClosableCardProps {
  id: string;
  badgeText: string;
  text: string;
  ctaText: string;
  ctaHref: string;
}

const classes = {
  container: 'molecule p-4 mt-6 rounded-lg bg-blue-900',
  button:
    'ml-auto -mx-1.5 -my-1.5 bg-blue-50 text-blue-900 rounded-lg focus:ring-2 focus:ring-blue-400 p-1 hover:bg-blue-200 inline-flex h-6 w-6 dark:bg-blue-900 dark:text-blue-400 dark:hover:bg-blue-800',
  ctaText: 'text-sm underline text-blue-400 hover:text-blue-300',
  text: 'mb-3 text-sm text-blue-400',
  topRow: 'flex items-center mb-3',
};

const ClosableCard: React.FC<ClosableCardProps> = ({
  id,
  badgeText,
  ctaHref, ctaText, text,
}) => {
  const { translatorAdapter } = useConfig();
  const t = translatorAdapter.translate();

  return (
    <div id={id} className={classes.container} role="alert">
      <div className={classes.topRow}>
        <Badge text={badgeText} />
        <button type="button" className={classes.button} data-collapse-toggle={id}>
          <CloseIcon />
        </button>
      </div>
      <p className={classes.text}>{t(text)}</p>
      <a className={classes.ctaText} href={ctaHref}>
        {t(ctaText)}
      </a>
    </div>
  );
};

export default ClosableCard;
