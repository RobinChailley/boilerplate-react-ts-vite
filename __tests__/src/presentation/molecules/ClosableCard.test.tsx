import { ConfigContext } from '@context/ConfigurationContext';
import ClosableCard from '@presentation/molecules/ClosableCard';
import renderer from 'react-test-renderer';

import getConfigContextValue from '../../../mocks/context/getConfigurationContextValue';

describe('ClosableCard', () => {
  it('should render correctly', async () => {
    let render = renderer.create(
      <ConfigContext.Provider value={getConfigContextValue()}>
        <ClosableCard text={''} badgeText={''} ctaHref={''} ctaText={''} id={''} />
      </ConfigContext.Provider>,
    );

    const snapshot = render!.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
