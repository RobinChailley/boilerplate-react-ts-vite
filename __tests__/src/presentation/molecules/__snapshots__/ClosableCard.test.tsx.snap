// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClosableCard should render correctly 1`] = `
<div
  className="molecule p-4 mt-6 rounded-lg bg-blue-900"
  id=""
  role="alert"
>
  <div
    className="flex items-center mb-3"
  >
    <span
      className="atom text-sm font-semibold mr-2 px-2.5 py-0.5 rounded bg-orange-200 text-orange-900"
    />
    <button
      className="ml-auto -mx-1.5 -my-1.5 bg-blue-50 text-blue-900 rounded-lg focus:ring-2 focus:ring-blue-400 p-1 hover:bg-blue-200 inline-flex h-6 w-6 dark:bg-blue-900 dark:text-blue-400 dark:hover:bg-blue-800"
      data-collapse-toggle=""
      type="button"
    >
      <svg
        aria-hidden="true"
        className="w-4 h-4"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </div>
  <p
    className="mb-3 text-sm text-blue-400"
  />
  <a
    className="text-sm underline text-blue-400 hover:text-blue-300"
    href=""
  />
</div>
`;
